dm/syncer/ddl.go

```diff
@@ -245,18 +249,43 @@ func adjustCollation(tctx *tcontext.Context, ddlInfo *ddlInfo, statusVars []byte
 				tctx.L().Warn("not found charset default collation.", zap.String("originSQL", ddlInfo.originDDL), zap.String("charset", strings.ToLower(justCharset)))
 				return
 			}
-			tctx.L().Info("detect create database risk which use explicit charset and implicit collation, we will add collation by SHOW CHARACTER SET", zap.String("originSQL", ddlInfo.originDDL), zap.String("collation", collation))
+			tctx.L().Info("detect create database risk which use explicit charset and implicit collation, we will add collation by INFORMATION_SCHEMA.COLLATIONS", zap.String("originSQL", ddlInfo.originDDL), zap.String("collation", collation))
 		} else {
 			// has no charset and collation
 			// add collation by server collation from binlog statusVars
-			collation, _ = event.GetServerCollationByStatusVars(statusVars)
+			collation, _ = event.GetServerCollationByStatusVars(statusVars, idAndCollationMap)
```

Comment:
log error? and also check collation is not ""

Reply:
As @glorv's comment about 'This shouldn't happen in a normal binlog event, so if this actually happened, I think we can just skip here.' in https://github.com/pingcap/ticdc/pull/3575, we may skip or log in info level?

Reply:
skip and print an ERROR or DPANIC level log 🤔

Reply:
ok

Reply:
Yes, we can output an ERROR log here and skip.